File: swift_pico/scripts/map.cpp
#include <iostream>
#include <fstream>
#include <opencv2/opencv.hpp>
#include <opencv2/aruco.hpp>
#include <opencv2/core.hpp>
#include <opencv2/imgcodecs.hpp>
#include <opencv2/highgui.hpp>

using namespace std; 

class Arena {

    public:

        int width;
        int height;
        int obstacles;
        float total_area;
        std::string image_path;
        int detected_markers[4];

        Arena(const std::string & path) {

            width = 1000;
            height =1000;
            obstacles = 0;
            total_area = 0;
            image_path = path;
        
        }

        void text_file(){
            std::ofstream file("obstacles.txt");
            if (file.is_open()) {
                file << "Aruco ID: [";
                for (int i = 0; i < 4; ++i) {
                    file << detected_markers[i];
                    if (i < 3) {
                        file << ", "; 
                    }
                }
                file << "]";
                file << "\nObstacles: " << obstacles << "\n";
                file << "Area: " << total_area << "\n";
                file.close();
            } else {
                cerr << "Failed to open file for writing.\n";
            }
        }
};

int main() {

    std::string image_path;
    image_path = 'task1c_image.jpg';
    Arena arena(image_path);
    arena.text_file();
    return 0;

}

File: swift_pico/scripts/map.py
import cv2
import numpy as np
import cv2.aruco as aruco

class Arena:

    def __init__(self, image_path):
        self.width = 1000
        self.height = 1000
        self.image_path = image_path
        self.detected_markers = []
        self.obstacles = 0
        self.total_area = 0

    def identification(self):

        # Read the image
        frame = cv2.imread()
        



        ###################################
        # Identify the Aruco ID's in the given image

        

        self.detected_markers = []
        ###################################
        # Apply Perspeactive Transform



        transformed_image = ()
        ###################################
        # Use the transformed image to find obstacles and their area



        self.total_area = 0
        ###################################


    def text_file(self):
        with open("obstacles.txt", "w") as file:
            file.write(f"Aruco ID: {self.detected_markers}\n")
            file.write(f"Obstacles: {self.obstacles}\n")
            file.write(f"Area: {self.total_area}\n")


if __name__ == '__main__':
    image_path = 'task1c_image.jpg'
    arena = Arena(image_path)
    arena.identification()
    arena.text_file()
